Create AI assistant index as part of setup initialization (#1424)
wazuh/wazuh-indexer-plugins: docs/dev/plugins/setup.md

        String index
        String template
        +Index(String index, String template)

>         +setClient(Client client) IndexInitiali— including administrators, by designzer

Reviewer: ```suggestion
        +setClient(Client client) IndexInitializer
```

- A single reserved-id document (id `"wazuh-ai-assistant-settings"`) holding the assistant-wide settings and the field anonymization policy, under `field_policy`
- A single reserved-id document (`"credentials"`), owned entirely by Content Manager. The administrative AI assistant API never reads or returns this document.

Example documents:

```json
// Provider document
{ "name": "test-ai", "type": "anthropic", "base_url": "https://api.anthropic.com", "model": "claude-opus-4-6", "api_key": "enc:v1:SC/RyOIBkdm+kGl", "is_default": true, "updated_at": "2026-08-03T09:54:52.193Z" }

// Settings + field policy document (reserved id "wazuh-ai-assistant-settings")
{
  "privacy_default_on": false,
  "privacy_default_per_provider": {},
  "user_can_override": true,
  "field_policy": [
    { "field": "wazuh.agent.name", "action": "anonymize", "kind": "HOST" },
    { "field": "wazuh.agent.host.ip", "action": "anonymize", "kind": "IP" },
    { "field": "wazuh.agent.id", "action": "allow" }
  ]
}
```

#### Administrative AI assistant API

> `.wazuh-internal-state` is registered as an OpenSearch Security system index, so no role's index permissionscan access the documents of the index. So queries to this index depen on the Administrative API provided for it:

Reviewer: `permissionscan` missing a space, `depen` → `depend`, and the table needs a blank line before it or it renders as literal pipe-separated text.

```suggestion
`.wazuh-internal-state` is registered as an OpenSearch Security system index, so no role's index permissions can reach its documents. Every query to this index goes through the administrative API below:

| Endpoint | Method | Cluster permission | Backed by |
```

| Endpoint | Method | Cluster permission | Backed by |
| --- | --- | --- | --- |
| `/_plugins/_setup/ai_assistant/settings` | `GET` | `plugin:wazuh/ai_assistant/settings/read` | `GetAiAssistantSettingsAction` / `TransportGetAiAssistantSettingsAction`, `Operation.SETTINGS` |
| `/_plugins/_setup/ai_assistant/settings` | `PUT` | `plugin:wazuh/ai_assistant/settings/write` | `PutAiAssistantSettingsAction` / `TransportPutAiAssistantSettingsAction`, `Operation.SETTINGS` |
| `/_plugins/_setup/ai_assistant/providers` | `GET` | `plugin:wazuh/ai_assistant/settings/read` | same `Get*` action, `Operation.LIST_PROVIDERS` — `AiAssistantSettingsAdminIndex.listProviders()` |
| `/_plugins/_setup/ai_assistant/providers` | `POST` | `plugin:wazuh/ai_assistant/settings/write` | same `Put*` action, `Operation.PUT_PROVIDER`; body must carry the UUID `id` to create with |
| `/_plugins/_setup/ai_assistant/providers/{id}` | `PUT`, `DELETE` | `plugin:wazuh/ai_assistant/settings/write` | same, `Operation.PUT_PROVIDER` / `Operation.DELETE_PROVIDER` |

> `GET /ai_assistant/settings` returns the settings document flat providers are a separate resource, listed via `GET /ai_assistant/providers`, which excludes the two reserved document ids (`"wazuh-ai-assistant-settings"`, `"credentials"`). 

Reviewer: Two sentences run together, `flat` left dangling. This also covers what the later `GET /ai_assistant/settings` paragraph repeats, so that one can go.

```suggestion
`GET /ai_assistant/settings` returns the settings document's source as-is, with its fields flat at the root. Providers are a separate resource, listed via `GET /ai_assistant/providers`, which excludes the two reserved document ids (`"wazuh-ai-assistant-settings"` and `"credentials"`).
```

`PUT /ai_assistant/settings` always replaces the whole document: the caller sends the complete set of settings fields and the complete `field_policy` array on every write, not a partial diff 

`POST /ai_assistant/providers`'s body must include an `id` field, since other integrations depend on the document ending up with the exact id they sent: it must be a UUID, rejected with `400` when missing or malformed

`DELETE /ai_assistant/providers/{id}` returns `404` when no provider exists with that id, including on a repeated delete.

> `GET /ai_assistant/settings` returns the settings document's source as-is. 

Reviewer: Duplicates the earlier `GET /ai_assistant/settings` paragraph. Suggest removing this line.

Author: Done in b7ba4c8cd08bff32b2cef1429a35710ea047872b
